### PR TITLE
Fixes #159: The contents of the cell are now shown on the first row in the tooltip. 

### LINKED ISSUE #159: Favorites list: Tooltip should show the content of the cell under mouse cursor
In the favorites list of a simulation, only the name and description of the parameter under cursor is shown as tooltip. It would be of much better help if the content of the respective cell, e.g., the "Container" cell, would be shown.
tooltip

### PR DESCRIPTION
**Pull request by peterdew**
Fixes #159 

# Description

Added cellvalue of cell under mousecursor to the top of the tooltip.

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

- [ ] Integration tests
- [ ] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):
image
image

# Questions (if appropriate):

### CODE REVIEW

**Review by rwmcintosh — APPROVED** [review]

**Review by rwmcintosh — COMMENTED** [review]
@peterdew what do you think about adding a line to the second section of the tooltip?

So in your screenshot it does not just say `Concentration`, but instead

**Name**
Concentration

Similar to the other sections

**Review by rwmcintosh — CHANGES_REQUESTED** [review]
We decided to add this section header as either

**Name**
Concentration

or

**Parameter Name**
Concentration

**Comment by peterdew** [comment]
I felt it would be even nicer to keep the description where available and make a 3-line parameter section. 

**Parameter**
 Name
Description
--------------
**Other Section**
text

**Review by rwmcintosh — APPROVED** [review]

File: src/MoBi.UI/Services/ToolTipCreator.cs [hunk 1]
```diff
@@ -13,7 +14,7 @@ namespace MoBi.UI.Services
    public interface IToolTipCreator : OSPSuite.UI.Services.IToolTipCreator
    {
       SuperToolTip ToolTipFor(NotificationMessage notificationMessage);
-      SuperToolTip ToolTipFor(IMoBiParameterDTO parameterDTO);
+      SuperToolTip ToolTipFor(IMoBiParameterDTO parameterDTO, string cellValue = null);
```
Comment (msevestre) [inline src/MoBi.UI/Services/ToolTipCreator.cs:17]: this `cellValue` parameter name is very confusing to me. What is a cell in the context of a parameter? I think this should be renamed and ideally, an example added if the context is hard to explaijn

File: src/MoBi.UI/Services/ToolTipCreator.cs [hunk 2]
```diff
@@ -65,8 +69,31 @@ private static SuperToolTip addFormulaToolTip(FormulaBuilderDTO formula, SuperTo
...
       {
-         var toolTip = CreateToolTip(calculationMethod.Description, calculationMethod.CalculationMethodName);
+         return CreateToolTip(calculationMethod.Description, calculationMethod.CalculationMethodName);
+      }
+
+      private SuperToolTip createCellValueToolTip(string cellValue)
+      {
+         if (cellValue == string.Empty)
+            return CreateToolTip();
+
```
Comment (msevestre) [inline src/MoBi.UI/Services/ToolTipCreator.cs:79]: what's with the ...?

**Review by msevestre — COMMENTED** [review]
image

I Don't understand the ... at the beginning? What does this mean

**Comment by peterdew** [comment]
> image
> 
> I Don't understand the ... at the beginning? What does this mean

It gives an indication that the text in the grid cell is repeated. That text ends with ... when it is truncated due to lack of space. 
@PavelBal What do you think?

File: src/MoBi.UI/Services/ToolTipCreator.cs (hunk 2 above)
Comment (peterdew) [reply src/MoBi.UI/Services/ToolTipCreator.cs:79]: It's the implementation of what is discussed in another comment.

File: src/MoBi.UI/Services/ToolTipCreator.cs (hunk 1 above)
Comment (peterdew) [reply src/MoBi.UI/Services/ToolTipCreator.cs:17]: The tooltip is at a gridcell, and this cell has a value / text. The underlaying object is a parameter, but it has a lot of properties. You could argue that it is thus the property value that matters, but only the grid cell displays the value including the correct formatting. Hence, I chose the name cellValue

**Comment by rwmcintosh** [comment]
What about like this
image

image

**Comment by rwmcintosh** [comment]
I made the change to put the hovered column caption in the tooltip and also refactored and used that same tooltip creation technique in another grid parameter view.

**Pull request merged**